mastermind.rb

```diff
@@ -52,38 +52,42 @@ def play
 		current_turn = 1
 		display_board
 		guess(current_turn)
-		feedback(current_turn)
-		while current_turn < 12 && @solution != @board[current_turn]["guess"] #!winner(current_turn)
+		solu = @solution
+		feedback(current_turn, solu)
+		while current_turn < 12 && solu != @board[current_turn]["guess"] #!winner(current_turn)
 			current_turn += 1
 			display_board
 			guess(current_turn)
-			feedback(current_turn)
+			feedback(current_turn, solu)
 		end
 
-		if @board[current_turn]["guess"] == @solution
+		if @board[current_turn]["guess"] == solu
 			puts "Congratulations!!! You stopped the nuclear detonation!!"
 		else
 			puts "The bomb exploded!!!!"
 			puts "You Lose!"
 		end
 	end
 
-	def feedback(current_turn)
+	def feedback(current_turn, solution_arr_in)
+		# Tests for matching numbers and index, returns "b"
 		@board[current_turn]["guess"].each_with_index { |el, idx|
 			spot_open = feedback_el(current_turn)
-			if @solution[idx] == el && spot_open != false
+			if solution_arr_in[idx] == el && spot_open != false
 				@board[current_turn]["feedback"][spot_open] = "b"
 			end
 		}
 
-		solution_arr = @solution
-		current_arr = @board[current_turn]["guess"]
+		# tests for if the the number is in the solution, returns "w"
+		solution_arr = solution_arr_in
 		@board[current_turn]["guess"].each_with_index { |el, idx|
```

Comment:
Just a quick note that curly braces for iterative statements should only be used if it fits into one line otherwise it should be `do end`